includes/admin/controllers/class-onboarding-wizard.php

```diff
@@ -0,0 +1,432 @@
+		);
+	}
+
+	/**
+	 * Remove the Onboarding Wizard submenu page from the bD parent menu
+	 * since it is not necessary to show that link there.
+	 *
+	 * @since x.x
+	 *
+	 * @return void
+	 */
+	public function remove_menu() {
+		remove_submenu_page( 'wpbdp_admin', self::PAGE_SLUG );
+	}
+
+	/**
+	 * Adds custom classes to the existing string of admin body classes.
+	 *
+	 * The function appends a custom class to the existing admin body classes, enabling full-screen mode for the admin interface.
+	 *
+	 * @since x.x
+	 *
+	 * @param string $classes Existing body classes.
+	 * @return string Updated list of body classes, including the newly added classes.
+	 */
+	public function add_admin_body_classes( $classes ) {
+		return $classes . ' wpbdp-admin-full-screen';
+	}
+
+	/**
+	 * Checks if the Onboarding Wizard was skipped during the plugin's installation.
+	 *
+	 * @since x.x
+	 * @return bool True if the Onboarding Wizard was skipped, false otherwise.
+	 */
+	public function has_onboarding_been_skipped() {
+		return get_option( self::ONBOARDING_SKIPPED_OPTION, false );
+	}
+
+	/**
+	 * Marks the Onboarding Wizard as skipped to prevent automatic redirects to the wizard.
+	 *
+	 * @since x.x
+	 * @return void
+	 */
+	public function mark_onboarding_as_skipped() {
+		update_option( self::ONBOARDING_SKIPPED_OPTION, true, 'no' );
+	}
+
+	/**
+	 * Check if the current page is the Onboarding Wizard page.
+	 *
+	 * @since x.x
+	 *
+	 * @return bool True if the current page is the Onboarding Wizard page, false otherwise.
+	 */
+	public function is_onboarding_wizard_page() {
+		return WPBDP_App_Helper::is_admin_page( self::PAGE_SLUG );
+	}
+
+	/**
+	 * Checks if the plugin has already performed a redirect to avoid repeated redirections.
+	 *
+	 * @return bool Returns true if already redirected, otherwise false.
+	 */
+	private function has_already_redirected() {
+		if ( get_option( self::REDIRECT_STATUS_OPTION ) ) {
+			return true;
+		}
+
+		update_option( self::REDIRECT_STATUS_OPTION, WPBDP_VERSION, 'no' );
+		return false;
+	}
+
+	/**
+	 * Get the path to the Onboarding Wizard views.
+	 *
+	 * @since x.x
+	 *
+	 * @return string Path to views.
+	 */
+	public static function get_page_url() {
+		return self::$page_url;
+	}
+
+	/**
+	 * Set the URL to access the Onboarding Wizard's page.
+	 *
+	 * @return void
+	 */
+	private function set_page_url() {
+		self::$page_url = admin_url( 'admin.php?page=' . self::PAGE_SLUG );
+	}
+
+	/**
+	 * Get the path to the Onboarding Wizard views.
+	 *
+	 * @since x.x
+	 *
+	 * @return string Path to views.
+	 */
+	public function get_view_path() {
+		return $this->view_path;
+	}
+
+	/**
+	 * Validates if the Onboarding Wizard page is being displayed.
+	 *
+	 * @since x.x
+	 * @deprecated x.x
+	 *
+	 * @return bool True if the Onboarding Wizard page is displayed, false otherwise.
+	 */
+	public static function is_onboarding_wizard_displayed() {
+		_deprecated_function( __METHOD__, 'x.x' );
```

Comment:
Why are we deprecating a method that we just added?

LE: I don't see any usage of this method in the current PR, so we might be safe just removing the function directly.